### PR TITLE
 Add webdav COPY operation

### PR DESCRIPTION
**Pull request by individual-it**
Add webdav COPY operation #44

### CODE REVIEW

**Comment by PVince81** [comment]
Wow, cool!

Can you please also add a test ?
You could reuse this https://github.com/owncloud/pyocclient/blob/master/owncloud/test/test.py#L286 and make a test for copy. After the copy, check that both the source and target files still exist.

**Comment by individual-it** [comment]
added 4 tests: 
- test_copy_in_place
- test_copy_into_subdir
- test_copy_unicode
- test_copy_to_non_existing_dir

File: owncloud/test/test.py [hunk 1]
```diff
@@ -429,6 +429,124 @@ def test_move_to_non_existing_dir(self):
...
+            )
+        )
+        self.assertTrue(
+            self.client.mkdir(
+                self.test_root + 'subdir'
+            )
+        )
+        self.assertTrue(
+            self.client.copy(
+                self.test_root + u'दिलै होस छोरा होस.txt',
```
Comment (PVince81) [inline owncloud/test/test.py]: I wonder what this says :smile:

**Comment by individual-it** [comment]
It's a  nepali saying
दिलै होस छोरा होस - You need time for doing something in a good way. - Haste makes waste.

File: owncloud/test/test.py (hunk 1 above)
Comment (individual-it) [inline owncloud/test/test.py]: It's a nepali saying.
दिलै होस छोरा होस - You need time for doing something in a good way. - Haste makes waste.

**Comment by PVince81** [comment]
Looks good :+1: 
Thanks a lot!

File: owncloud/test/test.py (hunk 1 above)
Comment (PVince81) [inline owncloud/test/test.py]: I suppose this also applies well to software development

**Pull request merged**